File: 0501-0600/0530-Minimum Absolute Difference in BST/0530-Minimum Absolute Difference in BST.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
 class Solution {
public:
    int getMinimumDifference(TreeNode* root) {
        stack<TreeNode*> St;
        TreeNode* node = root;
        int minDiff = INT_MAX;
        TreeNode* prev = nullptr;
        while (node || !St.empty()) {
            if (node) {
                St.push(node);
                node = node->left;
            }
            else {
                node = St.top();
                if (prev) {
                    minDiff = min(minDiff, node->val - prev->val);
                }
                prev = node;
                St.pop();
                node = node->right;
            }
        }
        
        return minDiff;
    }
};

File: 0501-0600/0530-Minimum Absolute Difference in BST/0530-Minimum Absolute Difference in BST.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

import math
class Solution:
    def getMinimumDifference(self, root: TreeNode) -> int:
        St = []
        minDiff = math.inf
        prev = None
        node = root
        while St or node:
            if node:
                St.append(node)
                node = node.left
            else:
                node = St.pop()
                if prev:
                    minDiff = min(minDiff, node.val - prev.val)
                prev = node
                node = node.right
        return minDiff
                